sql/expression/function/json/json_insert.go

```diff
@@ -0,0 +1,141 @@
+//     not an array, it is autowrapped as an array, then extended with the new value.
+//
+// Otherwise, a path-value pair for a nonexisting path in the document is ignored and has no effect.
+//
+// https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-insert
+type JSONInsert struct {
+	doc      sql.Expression
+	pathVals []sql.Expression
+}
+
+var _ sql.FunctionExpression = JSONInsert{}
+
+func (j JSONInsert) Resolved() bool {
+	for _, child := range j.Children() {
+		if child != nil && !child.Resolved() {
+			return false
+		}
+	}
+	return true
+}
+
+func (j JSONInsert) String() string {
+	children := j.Children()
+	var parts = make([]string, len(children))
+
+	for i, c := range children {
+		parts[i] = c.String()
+	}
+
+	return fmt.Sprintf("%s(%s)", j.FunctionName(), strings.Join(parts, ","))
+}
+
+func (j JSONInsert) Type() sql.Type {
+	return types.JSON
+}
+
+func (j JSONInsert) IsNullable() bool {
+	for _, arg := range j.pathVals {
+		if arg.IsNullable() {
+			return true
+		}
+	}
+	return j.doc.IsNullable()
+}
+
+func (j JSONInsert) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	doc, err := getMutableJSONVal(ctx, row, j.doc)
+	if err != nil || doc == nil {
+		return nil, err
+	}
+
+	pairs := make([]pathValPair, 0, len(j.pathVals)/2)
+	for i := 0; i < len(j.pathVals); i += 2 {
+		argPair, err := buildPathValue(ctx, j.pathVals[i], j.pathVals[i+1], row)
+		if argPair == nil || err != nil {
+			return nil, err
+		}
+		pairs = append(pairs, *argPair)
+	}
+
+	// Apply the path-value pairs to the document.
+	for _, pair := range pairs {
+		doc, _, err = doc.Insert(ctx, pair.path, pair.val)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return doc, nil
+}
+
+func (j JSONInsert) Children() []sql.Expression {
+	return append([]sql.Expression{j.doc}, j.pathVals...)
+}
+
+func (j JSONInsert) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(j.Children()) != len(children) {
+		return nil, fmt.Errorf("json_replace did not receive the correct amount of args")
+	}
+	return NewJSONInsert(children...)
+}
+
+// NewJSONInsert creates a new JSONInsert function.
+func NewJSONInsert(args ...sql.Expression) (sql.Expression, error) {
+	if len(args) <= 1 {
+		return nil, sql.ErrInvalidArgumentNumber.New("JSON_INSERT", "more than 1", len(args))
+	} else if (len(args)-1)%2 == 1 {
+		return nil, sql.ErrInvalidArgumentNumber.New("JSON_INSERT", "even number of path/val", len(args)-1)
+	}
+
+	return JSONInsert{args[0], args[1:]}, nil
+}
+
+// FunctionName implements sql.FunctionExpression
+func (j JSONInsert) FunctionName() string {
+	return "json_insert"
+}
+
+// Description implements sql.FunctionExpression
+func (j JSONInsert) Description() string {
+	return "inserts data into JSON document"
+}
+
+// IsUnsupported implements sql.UnsupportedFunctionStub
+func (j JSONInsert) IsUnsupported() bool {
+	return false
+}
```

Comment:
(minor) just fyi, not hurting anything, but I think you can completely omit this function if something is supported. It looks like the only place we reference this type in GMS is in `generate_readme.go` and a type must implement this interface and return true from this function in order for it to be marked as not supported, so shouldn't have any other effect if we omit this.